tests/signal.test.ts

```diff
@@ -0,0 +1,133 @@
+
+  try {
+    await client.rawRequest(ctx.url, `{ me { id } }`)
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort batch requests when the signal is defined using GraphQLClient constructor', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  const client = new GraphQLClient(ctx.url, { signal: abortController.signal })
+
+  try {
+    await client.batchRequests([{ document: `{ me { id } }` }, { document: `{ me { id } }` }])
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort a request when the signal overrides GraphQLClient settings', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  const client = new GraphQLClient(ctx.url)
+
+  try {
+    await client.request(ctx.url, `{ me { id } }`, undefined, abortController.signal)
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort a raw request when the signal overrides GraphQLClient settings', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  const client = new GraphQLClient(ctx.url)
+
+  try {
+    await client.rawRequest(ctx.url, `{ me { id } }`, undefined, abortController.signal)
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort batch requests when the signal overrides GraphQLClient settings', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  const client = new GraphQLClient(ctx.url)
+
+  try {
+    await client.batchRequests(
+      [{ document: `{ me { id } }` }, { document: `{ me { id } }` }],
+      undefined,
+      abortController.signal
+    )
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort a request', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  try {
+    await request(ctx.url, `{ me { id } }`, undefined, undefined, abortController.signal)
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort a raw request', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
+  expect.assertions(1)
+
+  try {
+    await rawRequest(ctx.url, `{ me { id } }`, undefined, undefined, abortController.signal)
+  } catch (error) {
+    expect((error as Error).message).toEqual('The user aborted a request.')
+  }
+})
+
+it('should abort batch requests', async () => {
+  const abortController = new AbortController()
+  abortController.abort()
```

Comment:
What happens if you abort after the request is sent but before the response is received?

Reply:
Same error is thrown before the response is received: `The user aborted a request.`
I've added a few tests for this cases in commit https://github.com/prisma-labs/graphql-request/pull/303/commits/53cd549336e54c17a4947dca3b59703dc7a23a3d